qcodes/dataset/data_set.py

```diff
@@ -262,6 +264,24 @@ def table_name(self):
     def guid(self):
         return get_guid_from_run_id(self.conn, self.run_id)
 
+    @property
+    def snapshot(self):
```

Comment:
Can we add minimal types (Optional[dict] and Optional[str]) to the return of these 2 functions.

Reply:
sure!